bioptim/gui/graph.py

```diff
@@ -0,0 +1,475 @@
+            The list of Mayer objectives
+        """
+
+        list_mayer_objectives = []
+        for objective in objective_list:
+            for obj_index in objective:
+                obj = obj_index["objective"]
+                if isinstance(obj.type, ObjectiveFcn.Mayer):
+                    mayer_str = ""
+                    mayer_objective = [obj.node[0]]
+                    if obj.sliced_target is not None:
+                        if obj.quadratic:
+                            mayer_str += f"({obj.name} - {obj.sliced_target})<sup>2</sup>"
+                        else:
+                            mayer_str += f"{obj.name} - {obj.sliced_target}"
+                    else:
+                        if obj.quadratic:
+                            mayer_str += f"({obj.name})<sup>2</sup>"
+                        else:
+                            mayer_str += f"{obj.name}"
+                    mayer_str = self._add_extra_parameters_to_str(obj.params, mayer_str)
+                    found = False
+                    for mayer in list_mayer_objectives:
+                        if mayer[1] == mayer_str:
+                            found = True
+                    if not found:
+                        mayer_objective.append(mayer_str)
+                        list_mayer_objectives.append(mayer_objective)
+        return list_mayer_objectives
+
+    def _scaling_parameter(self, parameter: Parameter):
+        """
+        Take scaling into account for display task
+
+        Parameters
+        ----------
+        parameter: Parameter
+            The unscaled parameter
+        """
+
+        initial_guess = [
+            parameter.initial_guess.init[i][j] * parameter.scaling[i]
+            for i in range(parameter.size)
+            for j in range(len(parameter.initial_guess.init[0]))
+        ]
+        min_bound = [
+            parameter.bounds.min[i][j] * parameter.scaling[i]
+            for i in range(parameter.size)
+            for j in range(len(parameter.bounds.min[0]))
+        ]
+        max_bound = [
+            parameter.bounds.max[i][j] * parameter.scaling[i]
+            for i in range(parameter.size)
+            for j in range(len(parameter.bounds.max[0]))
+        ]
+        return initial_guess, min_bound, max_bound
+
+    def _get_parameter_function_name(self, parameter: Parameter):
+        """
+        Get parameter function name (whether or not it is a custom function)
+
+        Parameters
+        ----------
+        parameter: Parameter
+            The parameter to which the function is linked
+        """
+        name = ""
+        if parameter.penalty_list is not None:
+            if parameter.penalty_list.type.name == "CUSTOM":
+                name = parameter.penalty_list.custom_function.__name__
+            else:
+                name = parameter.penalty_list.name
+        return name
+
+
+class OcpToConsole(GraphAbstratc):
+    _return_line = "\n"
+    _squared = "²"
```

Comment:
Will this work on Windows? Use the proper utf-8 symbol otherwise (or the less fun "^2")

Reply:
It works on Windows.

Reply:
Done